### DISCUSSION TITLE
ToStringWithData leading to high memory/CPU

### DISCUSSION THREAD

**Discussion by abko-auto** [Azure/azure-cosmos-dotnet-v3#3690] [General]
We had incident where our service in production was using CPU more than 60%. Upon closer look at memory, lot of allocation from ToStringWithData.
![image](https://user-images.githubusercontent.com/66053025/216804128-2beca980-68dc-4698-9e78-8f3488b43abe.png)

exception.ToString was active and exception had _innerException set so ToString was active ~6%.

![image](https://user-images.githubusercontent.com/66053025/216804299-6b5bf0b1-4d6d-4189-8d78-1b85a934d48e.png)

Mostly this logging I saw in Microsoft.Azure.Documents.GoneAndRetryWithRetryPolicy!ShouldRetryAsync function. There are lot of DefaultTrace.TraceError statements, I doubt they are collected if default listeners are removed. 

**Can we have configuration flag for verbose logging - If enabled emit these errors else not. This is consuming more than %10 of cpu + allocating those many string objects putting pressure on GC.**

**Comment by ealsur** [comment]
See https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3703
